addons/event/models/event_registration.py

```diff
@@ -390,6 +391,11 @@ def _message_post_after_hook(self, message, msg_vals):
 
     def _get_registration_summary(self):
         self.ensure_one()
+        if self.event_id.badge_format in ["96x82", "96x134"] and self.env.get("iot.device") is not None:
```

Comment:
🤮 
I know that the requirement to avoid adding a module comes from the internal team, to avoid installing yet another module in production, but making such "if the module is installed" is really something which must be avoided.

And later using `iot_websocket`, while it doesn't exist without the iot module. It only doesn't crash thanks to these `if` checking the existence of the module or not.

If there wasn't that requirement, it would be done in a separate module, `iot_event`, doing the overrides where needed to make this properly.

In addition, the module `iot` is not installed on odoo.com, so in any case this feature will require the installation of at least on module, `iot`, if we went this integration with iot printers.

Which is in any case adding a lot of models, and also an additional usage of the websocket, which I have no clue it has been load tested :s

Reply:
It looks like Olivier is in favor of that small if, even if not super clean, to avoid the creation and installation of a bridge module iot_event. So for the moment keep it like this.